packages/material-ui/src/useTouchRipple/useTouchRipple.ts

```diff
@@ -108,49 +113,54 @@ const useTouchRipple = (props: UseTouchRippleProps) => {
 
   const enableTouchRipple = mountedState && !disableRipple && !disabled;
 
-  const getRippleHandlers = React.useMemo(() => {
-    const rippleHandlers = {
-      onBlur: handleBlur,
-      onKeyDown: handleKeyDown,
-      onKeyUp: handleKeyUp,
-      onMouseDown: handleMouseDown,
-      onMouseUp: handleMouseUp,
-      onMouseLeave: handleMouseLeave,
-      onContextMenu: handleContextMenu,
-      onDragLeave: handleDragLeave,
-      onTouchStart: handleTouchStart,
-      onTouchEnd: handleTouchEnd,
-      onTouchMove: handleTouchMove,
-    } as RippleEventHandlers;
-
-    return (otherEvents: Partial<RippleEventHandlers> = {}) => {
-      const eventNames = Object.keys(rippleHandlers) as (keyof RippleEventHandlers)[];
-      const wrappedEvents = eventNames.map((eventName) => ({
-        name: eventName,
-        handler: (ev: any) => {
-          otherEvents[eventName]?.(ev);
-          rippleHandlers[eventName](ev);
-        },
-      }));
-
-      return wrappedEvents.reduce((acc, current) => {
-        acc[current.name] = current.handler;
-        return acc;
-      }, {} as RippleEventHandlers);
```

Comment:
This sure was clever. Though we can avoid creating extra data that nobody else needs, create more readable code, more debuggable code (can inspect specific event handlers) and type-safer code (specific events instead of `any`).